reconcile/ocm_labels/integration.py

```diff
@@ -0,0 +1,392 @@
+    def must_end_with_dot(  # pylint: disable=no-self-argument
+        cls, v: list[str]
+    ) -> list[str]:
+        return [prefix + "." if not prefix.endswith(".") else prefix for prefix in v]
+
+
+class ManagedLabelConflictError(Exception):
+    pass
+
+
+class OcmLabelsIntegration(QontractReconcileIntegration[OcmLabelsIntegrationParams]):
+    """Sync labels to subscription and organizations."""
+
+    @property
+    def name(self) -> str:
+        return QONTRACT_INTEGRATION
+
+    def get_clusters(self, query_func: Callable) -> list[ClusterV1]:
+        data = cluster_query(query_func)
+        return [
+            c
+            for c in data.clusters or []
+            if c.ocm is not None and integration_is_enabled(self.name, c)
+        ]
+
+    def get_organizations(
+        self, query_func: Callable
+    ) -> list[OpenShiftClusterManagerV1]:
+        return organization_query(query_func).organizations or []
+
+    def get_environments(self, query_func: Callable) -> list[OCMEnvironment]:
+        return ocm_environment_query(query_func).environments
+
+    def init_ocm_apis(
+        self,
+        environments: Iterable[OCMEnvironment],
+        init_ocm_base_client: Callable[
+            [OCMAPIClientConfigurationProtocol, SecretReaderBase], OCMBaseClient
+        ] = init_ocm_base_client,
+    ) -> dict[str, OCMBaseClient]:
+        return {
+            env.name: init_ocm_base_client(env, self.secret_reader)
+            for env in environments
+        }
+
+    def get_early_exit_desired_state(self) -> Optional[dict[str, Any]]:
+        # gqlapi = gql.get_api()
+        # return {"clusters": [c.dict() for c in self.get_clusters(gqlapi.query)]}
+        return {}
+
+    def run(self, dry_run: bool) -> None:
+        gqlapi = gql.get_api()
+        clusters = self.get_clusters(gqlapi.query)
+        organizations = self.get_organizations(gqlapi.query)
+        environments = self.get_environments(gqlapi.query)
+
+        self.ocm_apis = self.init_ocm_apis(environments, init_ocm_base_client)
+
+        # organization labels
+        orgs_current_state, orgs_desired_state = self.fetch_organization_label_states(
+            organizations
+        )
+        self.reconcile(
+            dry_run=dry_run,
+            scope="organization",
+            current_state=orgs_current_state,
+            desired_state=orgs_desired_state,
+        )
+
+        # subscription labels
+        subs_current_state, subs_desired_state = self.fetch_subscription_label_states(
+            clusters
+        )
+        self.reconcile(
+            dry_run=dry_run,
+            scope="cluster",
+            current_state=subs_current_state,
+            desired_state=subs_desired_state,
+        )
+
+    def fetch_organization_label_states(
+        self, organizations: Iterable[OpenShiftClusterManagerV1]
+    ) -> tuple[
+        dict[LabelOwnerRef, dict[str, str]], dict[LabelOwnerRef, dict[str, str]]
+    ]:
```

Comment:
Please define an alias for `tuple[dict[LabelOwnerRef, dict[str, str]], dict[LabelOwnerRef, dict[str, str]]]`. This would be easier to read.

Reply:
i created an alias for the inner part, so it reads `tuple[LabelState, LabelState]` now